File: 55-jump-game/55-jump-game.go
func canJump(nums []int) bool {
    if len(nums) == 1 {
        return true
    }
	total := len(nums)
	count := 0
	target := total - 2
	for i := target; i >= 0; i-- {
		count++
		if nums[i] >= count {
			target -= count
			count = 0
		}
	}
	if target < 0 {
		return true
	} else {
		return false
	}
}

File: 55-jump-game/55-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return True
        pointer = len(nums) - 2
        count = 1
        for i in range(2,len(nums)+1):
            if nums[len(nums)-i] >= count:
                pointer -= count
                count = 1
                continue
            count += 1

        if pointer < 0:
            return True
        else:
            return False
                